### PR TITLE
Participant password creation 

### PR DESCRIPTION
**Pull request by JoshAddington**
Adds changes to participant creation:
 - When participant registers, they are assigned an unusable password.
 - When study emails are sent out, selected participants:
   - are sent a forgotten password reset link if they already have a password from an earlier study.
   - have a password created for them and are sent that password if they do not already have a password.

### CODE REVIEW

File: project/myuser/email.py [hunk 1]
```diff
@@ -1,18 +1,39 @@
 from django.conf import settings
 from django.contrib.sites.models import Site
 from django.core.mail import send_mail
+from django.core.urlresolvers import reverse
 from django.template import Context
 from django.template.loader import render_to_string
 
 
+def generate_password_statement(participant):
```
Comment (shapiromatron) [inline project/myuser/email.py]: could we just pass the generated pin value or None to the welcome_email.* context, and then use the django template rendering for generating the text?

Comment (JoshAddington) [inline project/myuser/email.py]: might work

Comment (JoshAddington) [inline project/myuser/email.py]: I see you're being sneaky with your edits

Comment (shapiromatron) [inline project/myuser/email.py]: hah sorry, i thought slack would pick that up....

**Pull request merged**